crates/wasmtime/src/config.rs

```diff
@@ -1579,7 +1579,12 @@ impl Config {
             .insert("probestack_strategy".into(), "inline".into());
 
         let host = target_lexicon::Triple::host();
-        let target = self.compiler_config.target.as_ref().unwrap_or(&host);
+        let target = self
+            .compiler_config
+            .target
+            .as_ref()
+            .unwrap_or(&host)
+            .clone();
```

Comment:
I don't think this `.clone()` is necessary after the various changes this PR has gone through, though I could be wrong.

Reply:
Unfortunately seems like self.compiler_config.ensure_setting_unset_or_given still needs a mutable borrow while tripple would be borrowed immutability.

Reply:
Oops, you're right. We could fix that by moving the `ensure_setting_unset_or_given` method to `compiler_config.settings`, but cloning the triple isn't so bad either, so this is fine as-is.